pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,280 @@
+
+        # Create the initial state.
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+    def _create_basis_state(self, index):
+        """Return the density matrix representing a computational basis state over all wires.
+
+        Args:
+            index (int): integer representing the computational basis state.
+
+        Returns:
+            array[complex]: complex array of shape ``[2] * (2 * num_wires)``
+            representing the density matrix of the basis state.
+        """
+        rho = np.zeros((2 ** self.num_wires, 2 ** self.num_wires), dtype=np.complex128)
+        rho[index, index] = 1
+        rho = self._asarray(rho, dtype=self.C_DTYPE)
+        return self._reshape(rho, [2] * (2 * self.num_wires))
+
+    @property
+    def state(self):
+        """Returns the state density matrix of the circuit prior to measurement"""
+        dim = 2 ** self.num_wires
+        # User obtains state as a matrix
+        return self._reshape(self._pre_rotated_state, (dim, dim))
+
+    def reset(self):
+        """Resets the device"""
+        super().reset()
+
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+    def analytic_probability(self, wires=None):
+
+        if self._state is None:
+            return None
+
+        # convert rho from tensor to matrix
+        rho = self._reshape(self._state, (2 ** self.num_wires, 2 ** self.num_wires))
+        # probs are diagonal elements
+        probs = self.marginal_prob(self._diag(rho), wires)
+        return probs
+
+    def _get_kraus(self, operation):  # pylint: disable=no-self-use
+        """Return the Kraus operators representing the operation.
+
+        Args:
+            operation (~.Operation): a PennyLane operation
+
+        Returns:
+            list[array[complex]]: Returns a list of 2D matrices representing the Kraus operators. If
+            the operation is unitary, returns a single Kraus operator. In the case of a diagonal
+            unitary, returns a 1D array representing the matrix diagonal.
+        """
+        if isinstance(operation, DiagonalOperation):
+            return operation.eigvals
+
+        if isinstance(operation, Channel):
+            return operation.kraus_matrices
+
+        return [operation.matrix]
+
+    def _apply_channel(self, kraus, wires):
+        r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+        quantum state. For a unitary gate, there is a single Kraus operator.
+
+        Args:
+            matrices (array): Kraus operators
+            wires (Wires): target wires
+        """
+
+        channel_wires = self.map_wires(wires)
+
+        # Computes K^\dagger , which are needed for the transformation K \rho K^\dagger
+        kraus_dagger = [self._conj(self._transpose(k)) for k in kraus]
+
+        # Changes tensor shape
+        kraus_shape = len(kraus) + [2] * len(channel_wires) * 2
```

Comment:
Would this line work? :thinking: getting `TypeError: unsupported operand type(s) for +: 'int' and 'list'`

Reply:
Good catch! Yes, it should be `kraus_shape = [len(kraus)] + [2] * len(channel_wires) * 2`